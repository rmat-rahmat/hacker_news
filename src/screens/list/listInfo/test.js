/**
 * @format
 */

import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import ListInfo from '.';


describe('Test Details screen', () => {

const mockData = {
  "by": "Tester",
  "descendants": 71,
  "id": 1,
  "kids": [2, 20],
  "score": 111,
  "time": 1175714200,
  "title": "This is test story",
  "type": "story",
  "url": "http://www.getdropbox.com/u/2/screencast.html"
}


it('renders ListInfo correctly', () => {
  const tree = renderer.create(
    <ListInfo data={mockData} />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
});