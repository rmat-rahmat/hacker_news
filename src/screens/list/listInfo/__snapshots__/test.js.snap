// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Details screen renders ListInfo correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "flex": 0.1,
        "fontSize": 15,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 0.9,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
        }
      }
    >
      This is test story
       
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 15,
          }
        }
      >
        (
        getdropbox.com
        )
      </Text>
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
        }
      }
    >
      111
       points || by 
      Tester
       || 
      15 years ago
    </Text>
    <View
      style={
        Object {
          "color": "black",
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessibilityLabel="view comments"
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#841584",
                },
              ]
            }
          >
            71 comments
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
